### ISSUE TITLE
[Python][Pandas] (pandas 2.2.0) DateTimeTZBlock deprecation warning

### ISSUE THREAD

**Issue by Tishj** [apache/arrow#39724]
### Describe the bug, including details regarding any error messages, version, and platform.

When converting a pyarrow table containing timezone types, a deprecation warning is thrown

```py
import warnings
warnings.resetwarnings()

import pyarrow as pa
import datetime

dt = datetime.datetime(2023, 8, 29, 1, tzinfo=datetime.timezone.utc)

my_arrow_table = pa.Table.from_pydict({'ts': [dt, dt, dt], 'value': [1, 2, 3]})
df = my_arrow_table.to_pandas()
```

### Component(s)

Python

**Comment by jorisvandenbossche** [comment]
This is already fixed by https://github.com/apache/arrow/pull/38321, and this fix is included in the 15.0 release (this was just released yesterday, packages on PyPI are already available, conda packages still have to be updated)

**Comment by Tishj** [comment]
Thanks for letting me know!
